server.go

```diff
@@ -2534,11 +2534,24 @@ func newServer(listenAddrs []string, db database.DB, chainParams *chaincfg.Param
 	if len(indexes) > 0 {
 		indexManager = indexers.NewManager(db, indexes, chainParams)
 	}
-	bm, err := newBlockManager(&s, indexManager, interrupt)
+
+	// Create a new block chain instance with the appropriate configuration.
+	chain, err := blockchain.New(&blockchain.Config{
+		DB:          s.db,
+		Interrupt:   interrupt,
+		ChainParams: s.chainParams,
+		TimeSource:  s.timeSource,
+		Notifications: func(notif *blockchain.Notification) {
```

Comment:
I'd suggest `notification` for the parameter name to be consistent with `handleNotifyMsg`